mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala

```diff
@@ -205,15 +233,21 @@ class LinearSVC @Since("2.2.0") (
       val costFun = new LinearSVCCostFun(instances, $(fitIntercept),
         $(standardization), bcFeaturesStd, regParamL2, $(aggregationDepth))
 
-      def regParamL1Fun = (index: Int) => 0D
-      val optimizer = new BreezeOWLQN[Int, BDV[Double]]($(maxIter), 10, regParamL1Fun, $(tol))
+      val optimizerAlgo = $(optimizer) match {
```

Comment:
I saw other libraries like ```sklearn.svm.linearSVC``` use squared hinge loss as the default loss function, it's differentiable and can be solved by LBFGS. What about merging #17645 firstly and then we can use LBFGS as the default solver as well? Otherwise, we would make behavior change if we make this switch after 2.2 release. cc @jkbradley